File: Beginner/Attendance (ATTND)/attendance.py
t = int(input())


while t:
    N = int(input())
    A = []
    B = []

    for i in range(N):
        X, Y = map(str,input().split())
        A.append(X)
        B.append(Y)

    for j in range(N):
        if A.count(A[j]) > 1:
            print(A[j] + " " + B[j])
        else:
            print(A[j])

    t = t-1

        

File: Beginner/Attendance (ATTND)/attendance.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	loop := scanner.Text()
	testcase, _ := strconv.Atoi(loop)

	for testcase > 0 {
		scanner.Scan()
		text := scanner.Text()
		firstnameMap := make(map[string]int)
		atttendances := []string{}
		totalAttendance, _ := strconv.Atoi(text)
		for totalAttendance > 0 {
			scanner.Scan()
			text := scanner.Text()
			names := strings.Split(text, " ")
			firstname := names[0]
			firstnameMap[firstname]++
			atttendances = append(atttendances, text)
			totalAttendance--
		}

		for _, fullname := range atttendances {
			names := strings.Split(fullname, " ")
			firstname := names[0]
			surname := names[1]
			if firstnameMap[firstname] > 1 {
				fmt.Println(firstname, surname)
			} else {
				fmt.Println(firstname)
			}
		}
		testcase--
	}
}
